lib/grpc/server.ex

```diff
@@ -58,10 +74,24 @@ defmodule GRPC.Server do
   @doc false
   @spec call(atom, GRPC.Server.Stream.t,
     tuple, atom) :: {:ok, GRPC.Server.Stream.t, struct} | {:ok, struct}
-  def call(service_mod, stream, {_, {req_mod, req_stream}, {res_mod, res_stream}} = _rpc, func_name) do
-    marshal_func = fn(res) -> service_mod.marshal(res_mod, res) end
-    unmarshal_func = fn(req) -> service_mod.unmarshal(req_mod, req) end
-    stream = %{stream | marshal: marshal_func, unmarshal: unmarshal_func}
+  def call(service_mod, %{marshal: marshal, unmarshal: unmarshal} = stream, {_, {req_mod, req_stream}, {res_mod, res_stream}} = _rpc, func_name) do
+    # Unless a middleware inserted a marshal function, use the one provided by service_mod
```

Comment:
I just wonder why you want to override marshal/unmarshal functions in a middleware. I think the (maybe?) default is enough in most cases.

Reply:
Since all middleware functions operates at the same level of their `GRPC.*` counterparts, those parameters being available means that if I swap with a new marshaller that's expected for it to be taken in account, check the `FooReplacer` example. 

In practice though, I don't think that's a common use case, but it felt a bit counterintuitive to not be taken in account at that point. 

Another problem though is that's not supported on client-side (because `Stub.call/6` doesn't take a stream as a parameter). I'm not sure it's worth the cost to make such changes there. 

[Go implementation](https://github.com/grpc/grpc-go/blob/d1cd1551aa4cd02197f5715f82f9a4df8e053bc7/interceptor.go#L26) addresses this case by providing to middlewares directly the unmarshalled request and response and thus do not allow such a capability. 

I think it makes the whole thing nicer and clearer but I'd have to make more changes to server and stub code. 

WDYT?